test/core/state/replication.test.js

```diff
@@ -110,21 +107,53 @@ test.serial('ReplicaSetState apply throws if it cannot detect the replica state
   const client1 = new MongoClientStub([
     { myState: 2 } // second waitUntilHealthy should pass
   ])
-  const server1 = new MongoServerState('replication-test-0:27000', client1)
+  const server1 = new ReplicaMemberMongoServerState({ hostname: 'replication-test-0:27000' }, client1)
 
   const client2 = new MongoClientStub([
     { myState: 3 } // waitUntilHealthy
   ])
-  const server2 = new MongoServerState('replication-test-1:27000', client2)
+  const server2 = new ReplicaMemberMongoServerState({ hostname: 'replication-test-1:27000' }, client2)
 
   const servers = [server1, server2]
 
-  const configuration = new ReplicationConfiguration({ replicaSetName, servers })
-  const state = new ReplicaSetState(configuration)
+  const state = new ReplicaSetState(servers, { replicaSetName })
 
   await t.throwsAsync(async () => {
     await state.apply()
   }, { message: 'replication-test: Replica set status unknown. Cannot apply configuration.' })
 
   t.is(state.replicaSetStatus, REPLICA_SET_STATUS.UNKNOWN)
 })
+
+test('replication configuration structure works as expected for non config servers', (t) => {
+  const servers = [
+    new ReplicaMemberMongoServerState({ hostname: 'mongod-0:0' }, {}),
```

Comment:
What about calling this`host` as mongo does? it's not a hostname since it contains the port.